examples/gno.land/r/morgan/guestbook/guestbook_test.gno

```diff
@@ -0,0 +1,131 @@
+package guestbook
+
+import (
+	"std"
+	"strings"
+	"testing"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+)
+
+func TestSign(t *testing.T) {
+	guestbook = avl.Tree{}
+	hasSigned = avl.Tree{}
+
+	std.TestSetRealm(std.NewUserRealm("g1user"))
+	Sign("Hello!")
+
+	std.TestSetRealm(std.NewUserRealm("g1user2"))
+	Sign("Hello2!")
+
+	res := Render("")
+	t.Log(res)
+	if !strings.Contains(res, "> Hello!\n>\n> _Written by g1user ") {
+		t.Error("does not contain first user's message")
+	}
+	if !strings.Contains(res, "> Hello2!\n>\n> _Written by g1user2 ") {
+		t.Error("does not contain second user's message")
+	}
+	if guestbook.Size() != 2 {
+		t.Error("invalid guestbook size")
+	}
+}
+
+func TestSign_FromRealm(t *testing.T) {
+	std.TestSetRealm(std.NewCodeRealm("gno.land/r/demo/users"))
+
+	defer func() {
```

Comment:
btw, we have `urequire` & `uassert` if you want to try to use them. Up to you though